Visual Studio 2017 compilation error

This new code from last commit causes error on visual studio 2017

> #if __cplusplus < 201703L
> namespace std {
> template<typename... Ts> struct make_void { typedef void type;};
> template<typename... Ts> using void_t = typename make_void<Ts...>::type;
> }

> E2626	alias template type "std::make_void<Ts...>::type" is incompatible with the previous type of "void" in the redeclaration of alias template "std::void_t" (declared at line 179 of "c:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.14.26428\include\type_traits")

Trying to comment this line causes different error

> C2679	binary '=': no operator found which takes a right-hand operand of type 'Test_SDL_Strings_ConcatStringsTest_Test::TestBody::<lambda_9427c55ed5f3ba2f27f9d5a3c27cb397>' (or there is no acceptable conversion)

This is not only VS problem.
I have the same issue on Ubuntu after updating g++ from 5.4 to 7.3

`premock.hpp:189:72: error: conflicting declaration of template ‘template<class ... Ts> using void_t = typename std::make_void::type’
 template<typename... Ts> using void_t = typename make_void<Ts...>::type;`
I compile my tests in C++14 mode

I changed the guard from
`#if __cplusplus < 201703L`
to
`#if __cplusplus <= 201103L`
and it helped.

--- a/premock.hpp
+++ b/premock.hpp
@@ -187,7 +187,7 @@ public:
 #        define PREMOCK_NEED_VOID_T
 #    endif
 #else
-#    if __cplusplus < 201703L  // look ma, standards!
+#    ifndef __cpp_lib_void_t
 #        define PREMOCK_NEED_VOID_T
 #    endif
 #endif

